src/screens/Ticketing/Tickets/use-recent-tickets.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  PreassignedFareProduct,
+  UserProfile,
+} from '../../../reference-data/types';
+import {UserProfileWithCount} from '../Purchase/Travellers/use-user-count-state';
+import {useEffect, useMemo, useReducer} from 'react';
+import {
+  listRecentFareContracts,
+  RecentFareContract,
+} from '../../../api/fareContracts';
+import {TariffZoneWithMetadata} from '../Purchase/TariffZones';
+import {findReferenceDataById} from '../../../reference-data/utils';
+import {
+  RemoteConfigContextState,
+  useRemoteConfig,
+} from '../../../RemoteConfigContext';
+
+export type RecentTicket = {
+  preassignedFareProduct: PreassignedFareProduct;
+  fromTariffZone: TariffZoneWithMetadata;
+  toTariffZone: TariffZoneWithMetadata;
+  userProfilesWithCount: UserProfileWithCount[];
+};
+
+type State = {
+  error: boolean;
+  loading: boolean;
+  recentFareContracts: RecentFareContract[];
+};
+
+const initialState: State = {
+  error: false,
+  loading: true,
+  recentFareContracts: [],
+};
+
+type Action =
+  | {type: 'FETCH'}
+  | {type: 'ERROR'}
+  | {type: 'SUCCESS'; data: RecentFareContract[]};
+
+type Reducer = (prevState: State, action: Action) => State;
+
+const reducer: Reducer = (prevState, action): State => {
+  switch (action.type) {
+    case 'FETCH':
+      return {
+        ...prevState,
+        loading: true,
+        error: false,
+      };
+    case 'ERROR':
+      return {
+        ...prevState,
+        loading: false,
+        error: true,
+      };
+    case 'SUCCESS':
+      return {
+        ...prevState,
+        loading: false,
+        error: false,
+        recentFareContracts: action.data,
+      };
+  }
+};
+
+const mapUsers = (
+  users: {[user_profile: string]: string},
+  userProfiles: UserProfile[],
+) =>
+  Object.entries(users)
+    .map(([profileId, count]) => ({
+      ...findReferenceDataById(userProfiles, profileId),
+      count: parseInt(count),
+    }))
+    .filter((u): u is UserProfileWithCount => 'id' in u)
+    .sort(
+      // Sort by user profile order in remote config
+      (u1, u2) =>
```

Comment:
We also have `sortBy` dependency from lodash.

Reply:
Aight. Will use that instead.

Reply:
I did look into using sortBy. I think for these situations using sortBy did not give more readable nor more compact code, as it could not be used in a chain, and temp variables had to be introduced.